src/altinn_model_publisher/resources/update.py

```diff
@@ -1,17 +1,23 @@
 """Resource module for update."""
 import asyncio
 
-from flask import Response
-from flask_restful import Resource
+from aiohttp.web import Response, View
 
-from altinn_model_publisher.service.altinn_service import update_altinn_models
+from altinn_model_publisher.service.altinn_service import (
+    update_altinn_models,
+    UPDATE_IN_PROGRESS,
+)
 
 
-class Update(Resource):
+class Update(View):
     """Class representing update resource."""
 
     @staticmethod
-    def post() -> Response:
+    async def post() -> Response:
         """Update file with altinn models."""
-        asyncio.run(update_altinn_models())
-        return Response("OK")
+        update_status = await asyncio.create_task(update_altinn_models())
+
+        if update_status == UPDATE_IN_PROGRESS:
```

Comment:
Gjøres dette pga. oppdatering pågår i mongo? Kunne status da vært f.eks. DB_UPDATE_IN_PROGRESS for å si det mer tydelig?

Reply:
Kanskje den ikke er nødvendig uansett, eller?

Reply:
Mongo oppdateringen går kjapt, det er strengt tatt nedlasting fra altinn som er i gang. Ville hindre at flere workers begynte å laste ned fra altinn samtidig, å lagre en liten statusmelding i mongo var eneste måten jeg fikk til å kommunisere det mellom workers på.